### PR TITLE
set owner when creating a client

### LINKED ISSUE #24: Apps should be assigned to the user that made them
Right now all apps are saved to a table without a user id for the user that created them. We should create a view for apps filtered by the author id.

### PR DESCRIPTION
**Pull request by sethvincent**
close #24

### CODE REVIEW

**Comment by sethvincent** [comment]
It looks like there's not a way to filter clients by owner in the hydra api https://www.ory.sh/docs/hydra/sdk/api#list-oauth-20-clients

We'll have to do that at the application level I guess.

**Comment by sethvincent** [comment]
related issue: https://github.com/ory/hydra/issues/505

**Comment by kamicut** [comment]
I think that's fine, it happens at the osm-teams server level before rendering to the client, so I don't think there's a security problem there. There might be a performance issue _maybe_ if there are hundreds of clients, but I doubt that it will be a problem right now.

**Comment by sethvincent** [comment]
Cool, I was thinking the same: filtering inside our api endpoint would work for now and we should revisit later if we start to get a lot of clients.

File: app/manage/client.js [hunk 1]
```diff
@@ -12,10 +12,12 @@ const manageId = serverRuntimeConfig.OSM_HYDRA_ID
...
  */
 async function getClients (req, res) {
+  const { session: { user_id } } = req
   let clients = await hydra.getClients()
 
-  // Remove first party app from list
-  let filteredClients = clients.filter(c => c.client_id !== manageId)
+  // Remove first party client from list & exclude clients the user does not own
+  let filteredClients = clients
+    .filter(c => c.client_id !== manageId && c.owner === user_id)
```
[the remark this answers is not in the document]
Comment (kamicut) [reply app/manage/client.js]: Not sure, maybe as an admin of osm-teams I want to see all apps? But I think that's an enhancement

**Comment by kamicut** [comment]
I tested this and it works

**Pull request merged**